src/js/diff.js

```diff
@@ -960,7 +960,13 @@ fluid.diff.arraysEqual = function (leftArray, rightArray) {
  *
  */
 fluid.diff.objectsEqual = function (leftObject, rightObject) {
-    if (typeof leftObject !== typeof rightObject) {
+    if (leftObject === null && rightObject === null) {
```

Comment:
You might find that the use of a function like "fluid.typeCode" can short-circuit a lot of the logic in this leg - https://github.com/fluid-project/infusion/blob/main/src/framework/core/js/Fluid.js#L401

Reply:
Perhaps not actually fluid.typeCode itself, mind, but a function *like* fluid.typeCode : P 
"Countries like Australia, such as New Zealand, Australia itself, South Africa, ...."

Reply:
I took a stab at it and also used it in the `leftValue` / `rightValue` comparison below.